odoo/addons/base/models/res_company.py

```diff
@@ -309,6 +309,10 @@ def write(self, values):
 
         res = super(Company, self).write(values)
 
+        # Archiving a company should also archive all of its branches
+        if values.get('active') is False:
+            self.child_ids.active = False
```

Comment:
What about children of the children though?

Reply:
Well, won't the write() be called recursively ? They're companies.